pkg/daemon/ceph/osd/volume.go

```diff
@@ -265,6 +266,19 @@ func (a *OsdAgent) initializeBlockPVC(context *clusterd.Context, devices *Device
 			metadataBlockPath = devices.Entries["metadata"].Config.Name
 		}
 
+		if _, ok := devices.Entries["waldata"]; ok {
+			waldataDev = true
+			waldataArg = append(waldataArg, []string{"--block.wal",
+				devices.Entries["waldata"].Config.Name,
+			}...)
+
+			crushDeviceClass := os.Getenv(oposd.CrushDeviceClassVarName)
```

Comment:
Why is the crushDeviceClass needed with the wal?

Reply:
after rebase and update the patchset, no this issue anymore